dlls/dxgi/factory.c

```diff
@@ -679,12 +722,31 @@ void qemu_dxgi_factory_CreateSwapChainForHwnd(struct qemu_syscall *call)
 {
     struct qemu_dxgi_factory_CreateSwapChainForHwnd *c = (struct qemu_dxgi_factory_CreateSwapChainForHwnd *)call;
     struct qemu_dxgi_factory *factory;
+    struct qemu_dxgi_device *device;
+    struct qemu_dxgi_output *output;
+    IDXGISwapChain1 *host;
+    struct qemu_dxgi_swapchain *obj;
+    uint64_t *host_buffers;
 
-    WINE_FIXME("Unverified!\n");
+    WINE_TRACE("\n");
     factory = QEMU_G2H(c->iface);
+    device = QEMU_G2H(c->device);
+    output = QEMU_G2H(c->output);
 
-    c->super.iret = IDXGIFactory5_CreateSwapChainForHwnd(factory->host, QEMU_G2H(c->device), QEMU_G2H(c->window),
-            QEMU_G2H(c->swapchain_desc), QEMU_G2H(c->fullscreen_desc), QEMU_G2H(c->output), QEMU_G2H(c->swapchain));
+    c->super.iret = IDXGIFactory5_CreateSwapChainForHwnd(factory->host, device ? (IUnknown *)device->host : NULL,
```

Comment:
Did you test this with a 64 bit or 32 bit application? It looks OK to me, but pitfalls are hidden with wrapping 32 to 64 bit. I usually develop with 32 bit client apps to find them early.

If you used 64 bit apitrace - a sane choice for d3d11 - I'd request to keep the FIXME("Unverified\n"); in the 32 to 64 case. It'll save a lot of pain later when there are mysterious crashes.

From a quick look at it swapchain_desc and fullscreen_desc should have the same layout in both cases.

Reply:
I tried this with both the 32-bit and 64-bit builds of apitrace. apitrace remaps all the pointers and uses the native bitsize version when calling the function, so I don't think the bitsize of the original app matters too much.

Reply:
hmm my email reply didn't seem to make it through.

Anyhow, yeah, the 32 bit apitrace matters, not where the apitrace log was recorded.